### DISCUSSION TITLE
< best practice > storage untuk DB on EC2

### DISCUSSION THREAD

**Discussion by n00bC0d3** [awsugid/forum#23] [Q&A]
kalau ada use case bikin multipile ec2 untuk bikin High Availbility Database ( misalkan use case pakai ec2, tidak mau RDS ) contohnya bikin Oracle RAC dengan 4 instance. 
Berarti kan kita bikin 4 ec2, nah, untuk storage database nya, paling bagus pakai apa ya ? 
a. EFS --> autoscaling storage size
b. EBS ( type io1 / io2 ) , karena perlu multi attach dan IOPS nya bakal gede untuk database --> storage size nya gmn ?
c. any other suggestion ?

**Comment by rioastamal** [comment]
@n00bC0d3, Ini menarik, saya pribadi belum pernah menggunakan Oracle RAC. Saya pribadi sering menggunakan EFS untuk shared file system. Untuk Throughput scale dengan file-file besar memang cocok. Tapi jika untuk file-file kecil dengan jumlah masif malah performa jelas lebih baik EBS.

Beberapa hal yang perlu dikonsiderasi:

1. Pola Throughput scale read/write dari Oracle RAC sendiri seperti apa
2. Data Availability/Durability: Butuh multiple AZ atau cukup single AZ
3. EC2 Availability: Butuh multiple AZ atau single AZ? karena EBS hanya bisa di single AZ

Referensi:
- https://aws.amazon.com/efs/when-to-choose-efs/
- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volumes-multi.html
